src/System.Private.CoreLib/shared/System/Index.cs

```diff
@@ -20,17 +34,64 @@ public Index(int value, bool fromEnd)
             _value = fromEnd ? ~value : value;
         }
 
+        /// <summary>Create an Index pointing at first element.</summary>
+        public static Index Start => new Index(0);
```

Comment:
All these factory methods going through `Index(int, bool)` is bad for JIT. It would better to have a private Index(int) constructor and use that in these methods.

Reply:
ok, I'll do that

Reply:
by the way the second parameter of Index(int, bool) is optional now. I assume adding adding the constructor Index(int) would still be ok.

Reply:
Yes, though it's rather confusing. You have to be careful to call the right one (e.g. if you really need to call the one with the bool parameter you have to specify the parameter).

Reply:
@mikedn what you suggest to do with 

public static Index FromStart(int value) => new Index(value, false);

as we'll need to validate the parameter to be > 0. do you still suggest to use the private constructor and validate outside the private constructor? something like 

```C#
public static Index Start
{
    get
    {
            if (value < 0)
            {
                ThrowHelper.ThrowValueArgumentOutOfRange_NeedNonNegNumException();
            }
            return new Index(value);
    }
}

```

Reply:
I'll do that for now